src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecProc.java

```diff
@@ -24,27 +24,33 @@
 
     private static final Logger LOGGER = Logger.getLogger(ContainerExecProc.class.getName());
 
+    private final int pid;
     private final AtomicBoolean alive;
     private final CountDownLatch finished;
     private final ExecWatch watch;
     private final Callable<Integer> exitCode;
 
     /**
-     * 
+     *
+     * @param pid
      * @param watch
      * @param alive
      * @param finished
      * @param exitCode
-     *            a way to get the exit code
      */
-    public ContainerExecProc(ExecWatch watch, AtomicBoolean alive, CountDownLatch finished,
-            Callable<Integer> exitCode) {
+    public ContainerExecProc(int pid, ExecWatch watch, AtomicBoolean alive, CountDownLatch finished,
```

Comment:
backwards compatibility

Reply:
We can! Do we need to though? Do we expect any external code calling these?

Reply:
I'd rather keep it unless annotated with `Restricted` just in case

Reply:
old method needs to be added and deprecated